### PR TITLE
Fix #3584: Merge 4 versions of topic_practice_fragment.xml into a single xml file

### LINKED ISSUE #3584: Merge topic_practice_fragment.xml into single xml file
Currently there are 4 versions of `topic_practice_fragment.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by kritigupta45**
## Explanation
Fix #3584: merges the 4 versions of the `topic_practice_fragment.xml` file into a single XML file.

## Essential Checklist
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

**Comment by kritigupta45** [comment]
@rt4914 PTAL thanks.

**Comment by rt4914** [comment]
@ARJUPTA Can you please review this.

File: app/src/main/res/layout/topic_practice_fragment.xml [hunk 1]
```diff
@@ -13,9 +13,12 @@
     android:id="@+id/topic_practice_skill_list"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:background="@color/oppiaBackground"
     android:clipToPadding="false"
```
Comment (Arjupta) [inline app/src/main/res/layout/topic_practice_fragment.xml]: @rt4914 only sw600dp files have background color as `oppiaBackground`. Is it something that mock demands or we missed to add background color to mobile layout files.

**Review by Arjupta — APPROVED** [review]
@kritigupta45 nicely done. There is only one thing @rt4914 needs to answer other than that it looks good to me.

**Comment by Arjupta** [comment]
@kritigupta45 I just saw now that there are merge conflicts in this file. Can you resolve them before asking a review from @rt4914 .

**Comment by BenHenning** [comment]
Hi. As of today, some main reviewers have taken time off for the next few weeks, so it may take a little while before we can look at this PR. We appreciate your patience while some of our team members recharge. We'll be fully returning on 4 January 2021.

**Review by Arjupta — APPROVED** [review]
LGTM, only need @rt4914 to reply over this [comment](https://github.com/oppia/oppia-android/pull/4022#discussion_r765440507)

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -588,6 +588,11 @@
   <dimen name="section_title_text_view_layout_margin_top">28dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">32dp</dimen>
 
+  <!-- Topic Practice Fragment -->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Nit
`<!-- TopicPracticeFragment -->`

here and everywhere else in other files.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 3]
```diff
@@ -392,12 +392,15 @@
...
 
- <!-- SectionTitleFragment -->
+  <!-- SectionTitleFragment -->
   <dimen name="section_title_divider_view_layout_margin_top">48dp</dimen>
   <dimen name="section_title_text_view_layout_margin_start">120dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">32dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">120dp</dimen>
 
+  <!-- Topic Practice Fragment -->
+  <dimen name="topic_practice_fragment_padding_bottom">112dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml:402]: You have defined this here but there is not change in tablet files.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 4]
```diff
@@ -396,6 +396,9 @@
   <dimen name="section_title_text_view_layout_margin_top">32dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">96dp</dimen>
 
+  <!-- Topic Practice Fragment -->
+  <dimen name="topic_practice_fragment_padding_bottom">112dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml:400]: You have defined this here but there is not change in tablet files.

**Review by rt4914 — COMMENTED** [review]
You can also merge tablet files by copying `topic_practice_fragment.xml` file from `sw600dp-port` folder to `sw600dp` folder and then deleting the file from `sw600dp-port` and `sw600dp-land`

File: app/src/main/res/layout/topic_practice_fragment.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/layout/topic_practice_fragment.xml]: We missed it. It should be in both the files.

File: app/src/main/res/values/dimens.xml (hunk 2 above)
Comment (kritigupta45) [reply app/src/main/res/values/dimens.xml]: Done.

File: app/src/main/res/layout-sw600dp/topic_practice_fragment.xml [hunk 5]
```diff
@@ -23,4 +23,4 @@
     app:layout_constraintEnd_toEndOf="parent"
     app:layout_constraintStart_toStartOf="parent"
     app:layout_constraintTop_toTopOf="parent" />
-</layout>
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp/topic_practice_fragment.xml]: EOF is missing, add one blank line at the end.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@kritigupta45 There is just one nit change pending. Else everything is good. Thanks.

File: app/src/main/res/layout-sw600dp/topic_practice_fragment.xml (hunk 5 above)
Comment (kritigupta45) [reply app/src/main/res/layout-sw600dp/topic_practice_fragment.xml]: Done. PTAL.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**